### ISSUE TITLE
poetry-core 1.0.2 breaks dependency resolution for packages without the specified dependency version (*) in debug mode

### ISSUE THREAD

**Issue by kleschenko** [python-poetry/poetry#3663]
- [x] I am on the [latest](https://github.com/python-poetry/poetry/releases/latest) Poetry version.
- [x] I have searched the [issues](https://github.com/python-poetry/poetry/issues) of this repo and believe that this is not a duplicate.
- [x] If an exception occurs when executing a command, I executed it again in debug mode (`-vvv` option).

- **OS version and name**: 
- **Poetry version**: 1.1.4
- **Link of a [Gist](https://gist.github.com/) with the contents of your pyproject.toml file**: https://gist.github.com/kleschenko/77df4d568dff3722b61caae402b00b5a

## Issue

Poetry started to fail to lock dependencies containing `*` in their sub-dependency versions after the upgrade to `poetry-core==1.0.2`  when running in debug mode (`-vv`).

```
$ poetry lock -vv
Skipping virtualenv creation, as specified in config file.
Updating dependencies
Resolving dependencies...
   1: fact: poetry-test is 0.1.0
   1: derived: poetry-test
   1: fact: poetry-test depends on flake8-black (>=0.2.1,<0.3.0)
   1: selecting poetry-test (0.1.0)
   1: derived: flake8-black (>=0.2.1,<0.3.0)
   1: fact: flake8-black (==0.2.1) depends on flake8 (>=3.0.0)
   1: Version solving took 0.521 seconds.
   1: Tried 1 solutions.

  Stack trace:

  14  ~/.local/pipx/venvs/poetry/lib/python3.6/site-packages/clikit/console_application.py:131 in run
      status_code = command.handle(parsed_args, io)

  13  ~/.local/pipx/venvs/poetry/lib/python3.6/site-packages/clikit/api/command/command.py:120 in handle
      status_code = self._do_handle(args, io)

  12  ~/.local/pipx/venvs/poetry/lib/python3.6/site-packages/clikit/api/command/command.py:171 in _do_handle
      return getattr(handler, handler_method)(args, io, self)

  11  ~/.local/pipx/venvs/poetry/lib/python3.6/site-packages/cleo/commands/command.py:92 in wrap_handle
      return self.handle()

  10  ~/.local/pipx/venvs/poetry/lib/python3.6/site-packages/poetry/console/commands/lock.py:34 in handle
      return self._installer.run()

   9  ~/.local/pipx/venvs/poetry/lib/python3.6/site-packages/poetry/installation/installer.py:97 in run
      return self._do_install(local_repo)

   8  ~/.local/pipx/venvs/poetry/lib/python3.6/site-packages/poetry/installation/installer.py:229 in _do_install
      ops = solver.solve(use_latest=self._whitelist)

   7  ~/.local/pipx/venvs/poetry/lib/python3.6/site-packages/poetry/puzzle/solver.py:58 in solve
      packages, depths = self._solve(use_latest=use_latest)

   6  ~/.local/pipx/venvs/poetry/lib/python3.6/site-packages/poetry/puzzle/solver.py:235 in _solve
      self._package, self._provider, locked=locked, use_latest=use_latest

   5  ~/.local/pipx/venvs/poetry/lib/python3.6/site-packages/poetry/mixology/__init__.py:7 in resolve_version
      return solver.solve()

   4  ~/.local/pipx/venvs/poetry/lib/python3.6/site-packages/poetry/mixology/version_solver.py:84 in solve
      next = self._choose_package_version()

   3  ~/.local/pipx/venvs/poetry/lib/python3.6/site-packages/poetry/mixology/version_solver.py:401 in _choose_package_version
      self._add_incompatibility(incompatibility)

   2  ~/.local/pipx/venvs/poetry/lib/python3.6/site-packages/poetry/mixology/version_solver.py:442 in _add_incompatibility
      self._log("fact: {}".format(incompatibility))

   1  ~/.local/pipx/venvs/poetry/lib/python3.6/site-packages/poetry/mixology/version_solver.py:472 in _log
      self._provider.debug(text, self._solution.attempted_solutions)

  AttributeError

  'NoneType' object has no attribute 'group'

  at ~/.local/pipx/venvs/poetry/lib/python3.6/site-packages/poetry/puzzle/provider.py:701 in debug
      697│ 
      698│         if message.startswith("fact:"):
      699│             if "depends on" in message:
      700│                 m = re.match(r"fact: (.+?) depends on (.+?) \((.+?)\)", message)
    → 701│                 m2 = re.match(r"(.+?) \((.+?)\)", m.group(1))
      702│                 if m2:
      703│                     name = m2.group(1)
      704│                     version = " ({})".format(m2.group(2))
      705│                 else:
```

It looks like this is caused by the changed message format for dependencies without version:
- when using `poetry-core==1.0.0`: `fact: flake8-black (0.2.1) depends on black (*)`
- when using `poetry-core==1.0.2`: `fact: flake8-black (0.2.1) depends on black`

**Comment by kleschenko** [comment]
Looks like this issue has been resolved in the latest `poetry` release (1.1.5) and I cannot reproduce it anymore

**Comment by abn** [comment]
Thank you for following up @kleschenko
